pkg/kademlia/routing.go

```diff
@@ -166,6 +166,13 @@ func (rt *RoutingTable) ConnectionSuccess(node *pb.Node) error {
 		return nil
 	}
 
+	// Add self to routing table
```

Comment:
This should be correctly setup already at routing table startup.

Reply:
yes My PR is still wip

Reply:
The purpose of this was because yourself could be removed from your own routing table

Reply:
But it shouldn't modify `rt.self` in the process. `self` seems to be used throughout this package to mean "this is who I am"...

Reply:
I.e. implement `(rt *RoutingTable) UpdateSelf(node *pb.Node)` or similar rather than doing it through `ConnectionSuccess`

Reply:
yes self is added in NewRoutingTable. If for some reason something tries to remove it, we should add that as a restriction in removeNode

Reply:
created the function. Need to write a test now